casper/src/test/scala/coop/rchain/casper/genesis/GenesisTest.scala

```diff
@@ -62,169 +59,178 @@ class GenesisTest extends FlatSpec with Matchers with BeforeAndAfterEach with Bl
+      val bonds   = ProtoUtil.bonds(genesis)
+
+      log.infos.length should be(1)
+      validators
+        .map {
+          case (v, i) => Bond(ByteString.copyFrom(Base16.decode(v)), i.toLong)
+        }
+        .forall(
+          bonds.contains(_)
+        ) should be(true)
   }
 
-  it should "detect an existing bonds file in the default location" in {
-    val path      = genesisPath
-    val bondsFile = path.resolve("bonds.txt").toString
-    printBonds(bondsFile)
+  it should "parse the wallets file and include it in the genesis state" in withRawGenResources {
+    (runtime: Runtime, genesisPath: Path, log: LogStub[Id], time: LogicalTime[Id]) =>
+      val walletsFile = genesisPath.resolve("wallets.txt").toString
+      printWallets(walletsFile)
 
-    val runtime        = Runtime.create(storageLocation, storageSize)
-    val runtimeManager = RuntimeManager.fromRuntime(runtime)
-    val genesis = Genesis
-      .fromInputFiles[Id](None, numValidators, path, None, runtimeManager, rchainShardId, None)
-    runtime.close()
-    val bonds = ProtoUtil.bonds(genesis)
-
-    log.infos.length should be(1)
-    validators
-      .map {
-        case (v, i) => Bond(ByteString.copyFrom(Base16.decode(v)), i)
-      }
-      .forall(
-        bonds.contains(_)
-      ) should be(true)
+      val runtimeManager  = RuntimeManager.fromRuntime(runtime)
+      val _               = fromInputFiles()(runtimeManager, genesisPath, log, time)
+      val storageContents = StoragePrinter.prettyPrint(runtime.space.store)
+
+      walletAddresses.forall(storageContents contains _._1) should be(true)
   }
 
-  it should "parse the wallets file and include it in the genesis state" in {
-    val path        = genesisPath
-    val walletsFile = path.resolve("wallets.txt").toString
-    printWallets(walletsFile)
+}
 
-    val runtime        = Runtime.create(storageLocation, storageSize)
-    val runtimeManager = RuntimeManager.fromRuntime(runtime)
-    val _ = Genesis
-      .fromInputFiles[Id](None, numValidators, path, None, runtimeManager, rchainShardId, None)
-    val storageContents = StoragePrinter.prettyPrint(runtime.space.store)
-    runtime.close()
+object GenesisTest {
+  val storageSize     = 1024L * 1024
+  def storageLocation = Files.createTempDirectory(s"casper-genesis-test-runtime")
+  def genesisPath     = Files.createTempDirectory(s"casper-genesis-test")
+  val numValidators   = 5
+  val rchainShardId   = "rchain"
 
-    walletAddresses.forall(storageContents contains _._1) should be(true)
+  def fromInputFiles(
+      maybeBondsPath: Option[String] = None,
+      numValidators: Int = numValidators,
+      maybeWalletsPath: Option[String] = None,
+      minimumBond: Long = 1L,
+      maximumBond: Long = Long.MaxValue,
+      faucet: Boolean = false,
+      shardId: String = rchainShardId,
+      deployTimestamp: Option[Long] = Some(System.currentTimeMillis())
+  )(
+      implicit runtimeManager: RuntimeManager,
+      genesisPath: Path,
```

Comment:
It feels like the `genesisPath` and `runtimeManager` shouldn't be implicits. It looks like never make use of that fact.

Reply:
Yeah, that's fair. I wanted them to be implicit, but then couldn't figure out the multi-variable version of `implicit blockStore => ...` that gets used in the `BlockStore` test fixture.